### PR TITLE
Api: Fix returned message when enabling, disabling and deleting a non-existing user

### LINKED ISSUE #19988: HTTP API for deleting global user returning incorrect message for a non-existing user
**What happened**:
curl -X DELETE http://admin:admin@localhost:3000/api/admin/users/99 returns 200 "User deleted"

**What you expected to happen**:
404 Status Code and message: "User not found"

**How to reproduce it (as minimally and precisely as possible)**:
Try to remove non-existing user

**Anything else we need to know?**:
No

**Environment**:
- Grafana version: 6.2.2

### PR DESCRIPTION
**Pull request by dpavlos**
**What this PR does / why we need it**:
- Fixes #19988
- Applies the same logic for /api/admin/users/:id/disable and enable calls
- Adds unit tests for the case described in #19988 as well as for the enable/disable local user scenario

### CODE REVIEW

File: pkg/api/admin_users_test.go [hunk 1]
```diff
@@ -115,6 +155,26 @@ func TestAdminApiEndpoint(t *testing.T) {
 			So(userId, ShouldEqual, 42)
 		})
 	})
+
+	Convey("When a server admin attempts to delete a non-existing user", t, func() {
+		userId := int64(0)
```
Comment (aknuds1) [inline pkg/api/admin_users_test.go]:
```suggestion
		var userId int64
```

File: pkg/services/sqlstore/user.go [hunk 2]
```diff
@@ -523,6 +526,16 @@ func DeleteUser(cmd *models.DeleteUserCommand) error {
 }
 
 func deleteUserInTransaction(sess *DBSession, cmd *models.DeleteUserCommand) error {
+
```
Comment (aknuds1) [inline pkg/services/sqlstore/user.go]:
```suggestion
```

File: pkg/services/sqlstore/user.go [hunk 3]
```diff
@@ -523,6 +526,16 @@ func DeleteUser(cmd *models.DeleteUserCommand) error {
...
+	//Check if user exists
+	user := models.User{Id: cmd.UserId}
+	has, err := sess.Get(&user)
+	if err != nil {
+		return err
+	} else if !has {
```
Comment (aknuds1) [inline pkg/services/sqlstore/user.go]: More idiomatic I think:
```suggestion
	}
	if !has {
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Generally looks good to me, but there are some changes I'd like to see.

**Comment by dpavlos** [comment]
Thanks @aknuds1 for reviewing this. Do you prefer the suggested changes in one batch? I can squash the commits before merging this.

**Comment by aknuds1** [comment]
@dpavlos We squash when we merge so it isn't necessary :)

**Review by aknuds1 — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM

**Comment by aknuds1** [comment]
Merged! Thanks very much for contributing to Grafana!

**Pull request merged**